lib/enigma.rb

```diff
@@ -83,4 +107,48 @@ def create_shifts(key, date)
     split_key(key).zip(date_to_offsets(date)).map { |nums| nums.reduce(:+) }
   end
 
+  def find_key(encrypted_message, date)
+    key  = ""
+    offsets = date_to_offsets(date)
+    find_shifts(encrypted_message, date).each_with_index do |shift, index|
+      if ((shift - offsets[index]) % 27).to_s.length == 1 && index == 0
+        key.concat("0" + ((shift - offsets[index]) % 27).to_s)
+      elsif index == 0
+        key.concat(((shift - offsets[index]) % 27).to_s)
+      else
+        new_key_value = 0
+        until key.length > index + 1
+          if (key.chars.last.concat(new_key_value.to_s)).to_i % 27 == shift
+            key.concat((((new_key_value - offsets[index]) % 27).to_s))
+          end
+          new_key_value += 1
+        end
+      end
+    end
```

Comment:
This method is not working like I want it too. It only returns the correct key sometimes. I tried to find the key after finding the shifts and offsets but was having trouble if the key had too many numbers over 27.